File: backend/app/schemas/knowledge_base.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class KnowledgeBaseCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    description: str | None = Field(default=None, max_length=2000)


class KnowledgeBaseSummary(BaseModel):
    id: str
    name: str
    description: str | None = None

    model_config = ConfigDict(from_attributes=True)


class KnowledgeBaseRead(KnowledgeBaseSummary):
    document_count: int = 0
    chunk_count: int = 0
    status: str = "EMPTY"
    created_at: datetime
    updated_at: datetime


class KnowledgeDocumentRead(BaseModel):
    id: str
    knowledge_base_id: str
    original_filename: str
    file_type: str
    content_hash: str
    extracted_text_length: int
    status: str
    error_message: str | None = None
    chunk_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class KnowledgeSearchRequest(BaseModel):
    query: str = Field(min_length=1, max_length=1000)
    top_k: int | None = Field(default=None, ge=1, le=20)


class KnowledgeSearchItem(BaseModel):
    chunk_id: str
    document_name: str
    content_preview: str
    score: float


class KnowledgeSearchResponse(BaseModel):
    items: list[KnowledgeSearchItem]


class RagSource(BaseModel):
    knowledge_base_id: str
    knowledge_base_name: str
    document_name: str
    content_preview: str
    score: float

File: frontend/src/types/knowledgeBase.ts
export type KnowledgeBase = {
  id: string
  name: string
  description: string | null
  document_count: number
  chunk_count: number
  status: string
  created_at: string
  updated_at: string
}

export type KnowledgeBaseCreateRequest = {
  name: string
  description?: string
}

export type KnowledgeDocument = {
  id: string
  knowledge_base_id: string
  original_filename: string
  file_type: string
  content_hash: string
  extracted_text_length: number
  status: string
  error_message: string | null
  chunk_count: number
  created_at: string
  updated_at: string
}

export type KnowledgeSearchItem = {
  chunk_id: string
  document_name: string
  content_preview: string
  score: number
}

export type KnowledgeSearchResponse = {
  items: KnowledgeSearchItem[]
}
